### PR TITLE
Angular reactive form

### PR DESCRIPTION
**Pull request by Delian4o-web**

### CODE REVIEW

File: src/app/contact/contact.component.spec.ts [hunk 1]
```diff
@@ -1,25 +1,129 @@
...
   beforeEach(() => {
     fixture = TestBed.createComponent(ContactComponent);
     component = fixture.componentInstance;
+    component.ngOnInit();
     fixture.detectChanges();
   });
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
+  it("should call the onSubmit method", () => {
```
Comment (strizzwald) [inline src/app/contact/contact.component.spec.ts:39]: Good job on adding tests! 😄

File: src/app/contact/contact.component.spec.ts [hunk 2]
```diff
@@ -1,25 +1,129 @@
...
+    fixture.detectChanges();
+    spyOn(component, "onSubmit");
+    el = fixture.debugElement.query(By.css("button")).nativeElement;
+    el.click();
+
+    expect(component.onSubmit).toHaveBeenCalledTimes(1);
+  });
+
+  it("should test input errors", () => {
```
Comment (strizzwald) [inline src/app/contact/contact.component.spec.ts:47]: Consider renaming this test to "should make field required"

File: src/app/contact/contact.component.spec.ts [hunk 3]
```diff
@@ -1,25 +1,129 @@
...
+    component.registerForm.controls.lastName.setValue("Georgiev");
+    component.registerForm.controls.email.setValue("dgueorguiev9@gmail.com");
+    component.registerForm.controls.phoneNo.setValue("+27611411010");
+
+    let user: User;
+    component.userInfo.subscribe((value) => (user = value));
+
+    component.onSubmit();
+
+    console.log(user);
```
Comment (strizzwald) [inline src/app/contact/contact.component.spec.ts:105]: You can remove the `console.log` here.

File: src/app/contact/contact.component.ts [hunk 4]
```diff
@@ -1,15 +1,54 @@
...
+    private fb: FormBuilder,
+    private customValidator: CustomvalidationService
+  ) {}
 
-  constructor() { }
+  ngOnInit() {
+    this.registerForm = this.fb.group({
+      firstName: ["", Validators.required],
+      lastName: [""],
+      email: ["", [Validators.required, Validators.pattern("[^ @]*@[^ @]*")]],
```
Comment (strizzwald) [inline src/app/contact/contact.component.ts:26]: Consider also just using Angular's in-built email validator. I think you can add it here by adding `Validators.email` in your email form control

File: src/app/contact/contact.component.ts [hunk 5]
```diff
@@ -1,15 +1,54 @@
...
-  styleUrls: ['./contact.component.scss']
+  selector: "app-contact",
+  templateUrl: "./contact.component.html",
+  styleUrls: ["./contact.component.scss"],
 })
 export class ContactComponent implements OnInit {
+  registerForm: FormGroup;
+  submitted = false;
+
+  @Output() userInfo = new EventEmitter<User>();
```
Comment (strizzwald) [inline src/app/contact/contact.component.ts:15]: Do we need this EventEmitter?

Comment (Delian4o-web) [reply src/app/contact/contact.component.ts:15]: Nope hehe let me remove it

**Pull request merged**